### ISSUE TITLE
`@nx/vite:test` hangs after dependency upgrade, if multiple spec files

### ISSUE THREAD

**Issue by mattfysh** [nrwl/nx#22006]
### Current Behavior

On running `nx test foo` the vite test executor hangs, and you have to either CTRL+C to get out or wait for a "fetch timeout" error. I've included a reproducible shell script below for showing the problem and how it only occurs when there is more than one test file

After running the script below, the only way to fix is to manually downgrade the vite version used by vitest and vite-node, by editing the pnpm-lock.yaml file, replace all instances of `vite: 5.1.4(@types/node@20.11.20)` with `vite: 5.0.12(@types/node@20.11.20)` then run `pnpm i` - the test will begin working again

### Expected Behavior

The test should complete as expected

### Steps to Reproduce

```sh
# vitest problems

project_name="zzz"

npx create-nx-workspace@latest $project_name \
  --workspaceType integrated --ci skip --preset ts --packageManager pnpm

cd $project_name

nx g lib foo --bundler=none --unit-test-runner=vitest

nx test foo # 1 spec file - works
cp foo/src/lib/foo.spec.ts foo/src/lib/foo2.spec.ts
nx test foo # 2 spec files - works

pnpm update --latest

nx test foo # 2 spec files - fails
rm foo/src/lib/foo2.spec.ts
nx test foo # 1 spec files - works

pnpm i -D vite@5.0.12
cp foo/src/lib/foo.spec.ts foo/src/lib/foo2.spec.ts
nx test foo # 2 spec files - fails
```

### Nx Report

```shell
Node   : 21.6.2
OS     : darwin-arm64
pnpm   : 8.15.3

nx (global)        : 18.0.4
nx                 : 18.0.5
@nx/js             : 18.0.5
@nx/eslint         : 18.0.5
@nx/workspace      : 18.0.5
@nx/eslint-plugin  : 18.0.5
@nx/vite           : 18.0.5
@nx/web            : 18.0.5
typescript         : 5.3.3
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by mattfysh** [comment]
I'm going to chuck a bunch of other issues I've found in here as well, because it seems to me like the vite plugin is not being properly maintained

* without setting `watch` anywhere, it seems to be defaulting to `watch=true` which will cause all sorts of issues, e.g. snapshots won't update and teardown won't run. Despite the task ending after the first test run, the console output shows that it is running in watch mode
* setting options.watch to false will not change the setting (in either nx.json defaults or in project.json) - the only way to disable watch is to set it in vite.config.ts
* trying to run with the command in the docs that should be used to update snapshots does not work, the snapshots are not updated as if vitest was not run with the `-u` flag. The command from the docs is `nx run my-app:test -u`
* small nit on the docs: the links in the top-right of the page content are broken: https://nx.dev/nx-api/vite/executors/test

**Comment by getlarge** [comment]
@mattfysh I also had some trouble when migrating to vite ^5.0.0. and vitest ^1.0.0.
Did you try to tweak your Vite test config to include pool options?
```json
// ...
    pool: 'threads',
    poolOptions: {
      threads: {
        singleThread: true,
      },
    },
```

**Comment by andrei-shoreline** [comment]
I had the same issue with both `vitest@1.2.2` and `vitest@1.3.1` and the fix that I found is downgrading `vite` to `5.0.12`.

![image](https://github.com/nrwl/nx/assets/115982460/a75ac6be-17c0-4615-ba04-2c920e97a2c4)

For `yarn` I've added `resolutions` to `package.json` and regenerated `yarn.lock`
```json
  "resolutions": {
    "vite": "5.0.12"
  }
```

**Comment by ak274** [comment]
@mandarini upgrading vite to 5.1.x causing an issue it hangs the tests and then timeout which is also raised in vitest repo

https://github.com/vitest-dev/vitest/issues/5182

Is there any better workaround than above mentioned solution? Please let us know 🙏

**Comment by rafahoro** [comment]
Same issue here. Using ` pool: 'threads',` (singleThread) fixed it.
Thank you @getlarge and @mattfysh

**Comment by bheftel** [comment]
@mattfysh -
>     * trying to run with the command in the docs that should be used to update snapshots does not work, the snapshots are not updated as if vitest was not run with the `-u` flag. The command from the docs is `nx run my-app:test -u`
> 
>     * small nit on the docs: the links in the top-right of the page content are broken: https://nx.dev/nx-api/vite/executors/test

I opened a PR for the `-u` flag issue - https://github.com/nrwl/nx/pull/22185

New to contributing here, not sure if I should file a new issue for that or use this existing one

**Comment by DenizUgur** [comment]
`pool: "threads"` didn't helped me. Regardless of the settigns in `project.json` vitest starts in watch mode but `nx` exists after first run is completed. Seems like options are not being properly passed as pointed out by @bheftel

**Comment by berenddeboer** [comment]
> Same issue here. Using ` pool: 'threads',` (singleThread) fixed it. Thank you @getlarge and @mattfysh

pool threads is the default, so I can't see how this could make a difference.

**Comment by rafahoro** [comment]
> > Same issue here. Using ` pool: 'threads',` (singleThread) fixed it. Thank you @getlarge and @mattfysh
> 
> pool threads is the default, so I can't see how this could make a difference.

I don't know. May be is the 'singleThread' option that makes the difference?
Anyway, with that change, works for me.

**Comment by mattfysh** [comment]
The fix has been made available with the release of vite 5.1.6 - happy dependency upgrade day everybody! :tada:
